File: src/main/java/com/github/joelws/release/tracker/handler/ReleaseTrackerExceptionHandler.java
package com.github.joelws.release.tracker.handler;

import com.github.joelws.release.tracker.exception.ReleaseTrackerException;

import javax.ws.rs.core.Response;
import javax.ws.rs.ext.ExceptionMapper;

public class ReleaseTrackerExceptionHandler implements ExceptionMapper<ReleaseTrackerException> {

    @Override
    public Response toResponse(ReleaseTrackerException exception) {
        return Response.status(exception.getErrorResponse().getStatus()).entity(exception.getErrorResponse()).build();
    }
}

File: src/main/kotlin/com/github/joelws/release/tracker/handler/ReleaseTrackerExceptionHandler.kt
package com.github.joelws.release.tracker.handler

import com.github.joelws.release.tracker.exception.ReleaseTrackerException
import com.github.joelws.release.tracker.response.build
import javax.ws.rs.core.Response

import javax.ws.rs.ext.ExceptionMapper

open class ReleaseTrackerExceptionHandler : ExceptionMapper<ReleaseTrackerException> {

    override fun toResponse(exception: ReleaseTrackerException): Response = exception.restResponse.build()
}
